src/pybind/mgr/dashboard/tests/test_tools.py

```diff
@@ -35,10 +34,16 @@ def set(self, data, key):
         return dict(key=key, **data)
 
 
+@ApiController('foo/:key/:method')
+class FooResourceDetail(RESTController):
+    def list(self, key, method):
```

Comment:
I can see your use case for rbd snapshots. But having to define a new controller for a detail seems strange to me.

Reply:
I understand that it may be too troublesome to create a class just for a detail method, but at least it is possible. Since we are not actually using this kind of endpoint anywhere, is it worth to support a different way to declare a detail endpoint?

I can imagine having a decorator that you could use in your rest controller to declare detail endpoints. Something like:

```python
@ApiController('foo')
class Foo(RESTController):
  # ...
  @detail_endpoint(':key/:method')
  def detail(self, key, method):
    # ...
```

But I think we can add such decorator when we start to have cases like the above in our code.